cli/foundation-models/system/inference/image-text-embeddings/prepare_data.py

```diff
@@ -0,0 +1,234 @@
+import pandas as pd
+from zipfile import ZipFile
+import random
+import string
+
+
+def download_and_unzip(dataset_parent_dir: str) -> None:
+    """Download image dataset and unzip it.
+
+    :param dataset_parent_dir: dataset parent directory to which dataset will be downloaded
+    :type dataset_parent_dir: str
+    """
+    # Create directory, if it does not exist
+    os.makedirs(dataset_parent_dir, exist_ok=True)
+
+    # download data
+    download_url = "https://cvbp-secondary.z19.web.core.windows.net/datasets/image_classification/fridgeObjects.zip"
+    print(f"Downloading data from {download_url}")
+
+    # Extract current dataset name from dataset url
+    dataset_name = os.path.basename(download_url).split(".")[0]
+    # Get dataset path for later use
+    dataset_dir = os.path.join(dataset_parent_dir, dataset_name)
+
+    if os.path.exists(dataset_dir):
+        shutil.rmtree(dataset_dir)
+
+    # Get the name of zip file
+    data_file = os.path.join(dataset_parent_dir, f"{dataset_name}.zip")
+
+    # Download data from public url
+    urllib.request.urlretrieve(download_url, filename=data_file)
+
+    # extract files
+    with ZipFile(data_file, "r") as zip:
+        print("extracting files...")
+        zip.extractall(path=dataset_parent_dir)
+        print("done")
+    # delete zip file
+    os.remove(data_file)
+    return dataset_dir
+
+
+def read_image(image_path: str) -> bytes:
+    """Read image from path.
+
+    :param image_path: image path
+    :type image_path: str
+    :return: image in bytes format
+    :rtype: bytes
+    """
+    with open(image_path, "rb") as f:
+        return f.read()
+
+
+def prepare_data_for_online_inference(dataset_dir: str) -> None:
+    """Prepare request json files for online inference.
+
+    :param dataset_dir: dataset directory
+    :type dataset_dir: str
+    """
+    sample_image_1 = os.path.join(dataset_dir, "milk_bottle", "99.jpg")
+    sample_image_2 = os.path.join(dataset_dir, "can", "1.jpg")
+
+    # Generate sample request for image embeddings
+    image_request_json = {
+        "input_data": {
+            "columns": ["image", "text"],
+            "index": [0, 1],
+            "data": [
+                [
+                    base64.encodebytes(read_image(sample_image_1)).decode("utf-8"),
+                    "",
+                ],  # the "text" column should contain empty string
+                [base64.encodebytes(read_image(sample_image_2)).decode("utf-8"), ""],
+            ],
+        }
+    }
+
+    request_file_name = os.path.join(dataset_dir, "image_request_data.json")
+
+    with open(request_file_name, "w") as request_file:
+        json.dump(image_request_json, request_file)
+
+    # Generate sample request for text embeddings
+    text_request_json = {
+        "input_data": {
+            "columns": ["image", "text"],
+            "index": [0, 1],
+            "data": [
+                ["", "text sample 1"],  # the "text" column should contain empty string
```

Comment:
Would it make sense to use a text that is related to the dataset ? Maybe something like "A photo of a milk bottle" ? Same comment applies to the image and text embeddings example